pinot-common/src/thrift/query.thrift

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+namespace java org.apache.pinot.common.request
+
+struct DataSource {
+  1: optional string tableName;
+}
+
+struct PinotQuery {
+  1: optional i32 version;
+  2: optional DataSource dataSource;
+  3: optional list<Expression> selectList;
```

Comment:
A question on semantics: Will the groupByList now need to be contained in the selectList ?

I like the fact that we are not representing aggregation and non-aggregation functions separately. That would help us retain the order.

Reply:
It's not a must right now but we can enforce that.